tests/test_functors.py

```diff
@@ -42,119 +42,181 @@
 
     def testCustom(self):
+        self.columns.append("base_FootprintArea_value")
+        self.dataDict["base_FootprintArea_value"] = \
+            np.random.rand(self.nRecords)
+        parq = self.simulateMultiParquet(self.dataDict)
         func = CustomFunctor('2*base_FootprintArea_value', filt='HSC-G')
-        val = self._funcVal(func)
+        val = self._funcVal(func, parq)
 
         func2 = Column('base_FootprintArea_value', filt='HSC-G')
 
-        np.allclose(val.values, 2*func2(self.parq).values, atol=1e-13, rtol=0)
+        np.allclose(val.values, 2*func2(parq).values, atol=1e-13, rtol=0)
 
     def testCoords(self):
-        ra = self._funcVal(RAColumn())
-        dec = self._funcVal(DecColumn())
+        parq = self.simulateMultiParquet(self.dataDict)
+        ra = self._funcVal(RAColumn(), parq)
+        dec = self._funcVal(DecColumn(), parq)
 
         columnDict = {'dataset': 'ref', 'filter': 'HSC-G',
                       'column': ['coord_ra', 'coord_dec']}
-        coords = self.parq.toDataFrame(columns=columnDict) / np.pi * 180.
+        coords = parq.toDataFrame(columns=columnDict, droplevels=True) / np.pi * 180.
 
-        self.assertTrue(np.allclose(ra, coords['coord_ra'], atol=1e-13, rtol=0))
-        self.assertTrue(np.allclose(dec, coords['coord_dec'], atol=1e-13, rtol=0))
+        self.assertTrue(np.allclose(ra, coords[('ref', 'HSC-G', 'coord_ra')], atol=1e-13, rtol=0))
+        self.assertTrue(np.allclose(dec, coords[('ref', 'HSC-G', 'coord_dec')], atol=1e-13, rtol=0))
 
     def testMag(self):
+        self.columns.extend(["base_PsfFlux_instFlux", "base_PsfFlux_instFluxErr"])
+        self.dataDict["base_PsfFlux_instFlux"] = np.full(self.nRecords, 1000)
+        self.dataDict["base_PsfFlux_instFluxErr"] = np.full(self.nRecords, 10)
+        parq = self.simulateMultiParquet(self.dataDict)
         fluxName = 'base_PsfFlux'
 
         # Check that things work when you provide dataset explicitly
         for dataset in ['forced_src', 'meas']:
             psfMag_G = self._funcVal(Mag(fluxName, dataset=dataset,
-                                         filt='HSC-G'))
+                                         filt='HSC-G'),
+                                     parq)
             psfMag_R = self._funcVal(Mag(fluxName, dataset=dataset,
-                                         filt='HSC-R'))
+                                         filt='HSC-R'),
+                                     parq)
 
             psfColor_GR = self._funcVal(Color(fluxName, 'HSC-G', 'HSC-R',
-                                              dataset=dataset))
+                                              dataset=dataset),
+                                        parq)
 
             self.assertTrue(np.allclose((psfMag_G - psfMag_R).dropna(), psfColor_GR, rtol=0, atol=1e-13))
 
         # Check that behavior as expected when dataset not provided;
         #  that is, that the color comes from forced and default Mag is meas
```

Comment:
The following is not testing the same thing it used to be; when the test data was a partial real dataset, 'meas' and 'forced' were different, and then since the default 'dataset' when not explicitly provided were different for `Mag` and `Color`, the `np.allclose()` below returned `False` (as it used to be an `assertFalse`).  So either the behavior of `Mag` and `Color` have changed or the generated test dataset has identical 'meas' and 'forced' values.

Reply:
This seems like the test should be more of a general test of a generic functor loading/calculating on the correct dataset/filter. I've reverted it to fail again (thanks for catching it, didn't quite appreciate why it was supposed to fail). I can write the generic functor test in a future ticket if you would like me to make it. I have more tickets to write specific tests of values so I'll likely just leave things as simple values for now.

Reply:
sure, sounds good